base/broadcast.jl

```diff
@@ -216,36 +245,36 @@ function broadcast_t(f, ::Type{Any}, As...)
         return similar(Array{Union{}}, shape)
     end
     nargs = length(As)
-    keeps, Idefaults = map_newindexer(shape, As)
+    keeps, Idefaults = map_indexer(shape, As)
     st = start(iter)
     I, st = next(iter, st)
-    val = f([ As[i][newindex(I, keeps[i], Idefaults[i])] for i=1:nargs ]...)
+    val = f(( _broadcast_getindex(As[i], newindex(I, keeps[i], Idefaults[i])) for i=1:nargs )...)
```

Comment:
you're making a generator, then splatting it, and calling the function on the splatted result - isn't one of the generators and splats redundant?